### ISSUE TITLE
can you share the train/val/test video name in each dataset?

### ISSUE THREAD

**Issue by hugddygff** [Justin1904/Low-rank-Multimodal-Fusion#6]
Hi Justin,

since you give the 3-D and 2-D array directly, it is difficult to know the clear train/val/test set.

And I find some value in the feature array is inf, causing the training loss "NAN", is this normal ?

Thanks very much!

**Comment by Justin1904** [comment]
For the first question, I wonder what information are you looking for about the train/val/test set. Could you elaborate? In the current state the train/val/test set is three distinct arrays and I hope that isn't too confusing.

For the second question, this is "normal" in the sense that inf values do appear in acoustic features due to the issues with COVAREP feature extraction. We replaced them with 0s, but you can do whatever data imputation you like.

**Comment by hugddygff** [comment]
thanks for the quick reply,

For the first question, we take the cmumosi dataset as an example. according to the sdk， the feature can be obtained through “cmumosi_highlevel.computational_sequences[“glove_vectors”].data[video_name][“features”]” the keys of data are the video_name of the 93 videos in cmumosi dataset. Therefore， I ask that which 52 videos are uesd to train, same for valid and test.

Since the 2-d array and 3-d array have no vid name information, if I change the features extracted by myself, i do not know which 52 videos are used for training, same as valid and test. 

of course , directly using your feature array is ok, if I do not change features.

**Comment by Justin1904** [comment]
I see. In fact, you can obtain the data splits through the SDK:

```
train_ids = mmsdk.mmdatasdk.cmu_mosi..standard_folds.standard_train_fold
dev_ids = mmsdk.mmdatasdk.cmu_mosi..standard_folds.standard_valid_fold
test_ids = mmsdk.mmdatasdk.cmu_mosi..standard_folds.standard_test_fold
```

I believe that data split is the same as the one used here (and most previous work using MOSI dataset as well), so you can just use that split and it would still be a fair comparison to them.

**Comment by hugddygff** [comment]
Thanks very much! 
And I try some experiments on cmumosi, but my metrics are different from that of the paper. I got mae about 1.0, acc/f1 about 0.725, corr about 0.60, multiclass 0.32, is there any other tricks to get the better results?

**Comment by Justin1904** [comment]
I think for MOSI that is actually within a reasonable range. We did a lot of grid search to get to the final results and during the process I do remember having results in the range of acc. 68-76. Unfortunately we didn't document the best setting for MOSI, so you might need to grid search on it yourself, and during our experiments we did find that MOSI is the hardest dataset to get top performance :( On the other hand, we do document the model hyperparameter settings for POM (and also IEMOCAP if you have the permission from USC and want to run experiment on it). So it should be easier to start with those first. Some settings like early stopping scheme and learning rate schedules were not recorded but it should be relatively easy to replicate the results on POM and IEMOCAP with various settings on these. Unfortunately on MOSI this seems to be really tricky.

In general, since MOSI is a fairly small dataset, overfitting becomes a severe problem. So it might be worth trying regularizing your model through dropout, L2, early stopping, etc.

On Mar 12, 2019, at 9:40 PM, kas-one <notifications@github.com<mailto:notifications@github.com>> wrote:


Thanks very much!
And I try some experiments on cmumosi, but my metrics are different from that of the paper. I got mae about 1.0, acc/f1 about 0.725, corr about 0.60, multiclass 0.32, is there any other tricks to get the better results?

**Comment by Justin1904** [comment]
@hugddygff There're problems in the SDK's features AFAIK. I am sure its transcripts (and also glove vectors) have incorrect timestamps. As for visual and acoustic features, I haven't checked.

**Comment by hugddygff** [comment]
@Justin1904 I also think the features of CMUMOSI on SDK have problem, therefore I decide to use the old_processed_features, it can obtain normal results. And I have a new problem, since the old_processed_features of iemocap on http://immortal.multicomp.cs.cmu.edu/raw_datasets/old_processed_data/iemocap/, its label is an 11-dimensional one-hot label, however, the label of iemocap provided by you is different, it contains 4 emotions, each emotion has a 2-dimensional one-hot label. Can you share the details?

Thanks a lot.

**Comment by Justin1904** [comment]
I am not sure why the one on the immortal server is different. The labels I am using are the occurrences of 4 different emotions, happy, sad, angry and neutral. Namely, the four labels represent whether these emotions occurred in the video segment.
